e2e-test-server/go.mod

```diff
@@ -1,10 +1,10 @@
 module github.com/GoogleCloudPlatform/opentelemetry-operations-go/e2e-test-server
 
-go 1.16
+go 1.14
```

Comment:
Should we be downgrading the go version?

Reply:
I see the commit message above.  Can you provide more details on what happened?

Reply:
All our other go.mod files say 1.14. Because this one file says 1.16, my local Go (1.16) would always make a change in go.sum that would be rejected by our CI (which runs 1.14). I had to run `make precommit` under Go 1.14 using Docker to make CI happy.

Reply:
Thanks.